lib/string-checker.js

```diff
@@ -126,6 +133,41 @@ StringChecker.prototype = {
     },
 
     /**
+     * Fix provided error.
+     *
+     * @param {JsFile} file
+     * @param {Errors} errors
+     * @protected
+     */
+    _fixJsFile: function(file, errors) {
+        var list = errors.getErrorList();
+        var configration = this.getConfiguration();
+
+        list.forEach(function(error) {
+            if (error.fixed) {
+                return;
+            }
+
+            var instance = configration.getConfiguredRule(error.rule);
+
+            if (instance && instance.fix) {
+                try {
+
+                    // "error.fixed = true" should go first, so rule can
+                    // decide for itself (with "error.fixed = false")
+                    // if it can fix this particular error
+                    error.fixed = true;
```

Comment:
> if case is too complicated for example

My alternative implementation would be to have the rule throw an error in the fix method for unfixable errors. I guess an error being unfixable isn't an exception? What I'm getting at is avoiding rules knowing about the `fixed` property of the error.

Of course, `fixed` is public, so it doesn't matter so much.